pkgs/applications/editors/gnome-inform7/default.nix

```diff
@@ -0,0 +1,94 @@
+{ lib, stdenv, fetchFromGitHub, meson, ninja, pkg-config, python3, perl, bison
+, texinfo, desktop-file-utils, wrapGAppsHook, docbook2x, docbook_xml_xslt
+, inform7, gettext, libossp_uuid, gnome2, gnome3, webkit, gtkspell3, libxml2
+, goocanvas2, libplist, gobjectIntrospection, glib, gst_all_1 }:
+
+# Neither gnome-inform7 nor its dependencies ratify and chimara have tagged releases in the GTK3 branch yet.
+
+let
+  ratify = (stdenv.mkDerivation {
+    name = "ratify";
+    src = fetchFromGitHub {
+      owner = "ptomato";
+      repo = "ratify";
+      rev = "f4d2d60ec73d5588e953650b3879e69a727f30ca";
+      sha256 = "eRh/9pYvdfbdbdJQ7pYMLq5p91I+rtyb/AqEGfakjKs=";
+    };
+    nativeBuildInputs = [ meson ninja pkg-config docbook2x docbook_xml_xslt ];
+    buildInputs = [
+      gnome3.gtk
+      gobjectIntrospection
+      gnome3.vala
+      gnome2.gtk-doc
+      wrapGAppsHook
+    ];
+  });
+
+  chimara = (stdenv.mkDerivation {
+    name = "chimara";
+    src = fetchFromGitHub {
+      owner = "chimara";
+      repo = "Chimara";
+      rev = "9934b142af508c75c0f1eed597990f39495b1af4";
+      sha256 = "aRz1XX8XaSLTBIrMIIMS3QNMm6Msi+slrZ6+KYlyRMo=";
+    };
+    nativeBuildInputs =
+      [ meson ninja pkg-config perl bison texinfo python3 glib wrapGAppsHook ];
+    buildInputs = [
+      gnome3.gtk
+      gobjectIntrospection
+      gnome3.vala
+      gnome2.gtk-doc
+      gst_all_1.gstreamer
+      gst_all_1.gst-plugins-base
+      gst_all_1.gst-plugins-good
+      gst_all_1.gst-plugins-bad
+      glib
+    ];
+    preConfigure = ''
+      patchShebangs build-aux/meson_post_install.py
+    '';
+  });
+
+in stdenv.mkDerivation {
+  name = "gnome-inform7";
+  version = "unstable-2021-04-06";
+  src = fetchFromGitHub {
+    owner = "ptomato";
+    repo = "gnome-inform7";
+    rev = "c37e045c159692aae2e4e79b917e5f96cfefa66a";
+    sha256 = "Q4xoITs3AYXhvpWaABRAvJaUWTtUl8lYQ1k9zX7FrNw=";
+  };
+  nativeBuildInputs =
+    [ meson ninja pkg-config inform7 python3 desktop-file-utils wrapGAppsHook ];
+  buildInputs = [
+    gettext
+    libossp_uuid
+    gnome3.gtk
+    gnome3.gtksourceview
+    gnome3.gspell
+    webkit
+    gtkspell3
```

Comment:
Does it really use gspell and gtkspell at the same time?

Reply:
No, it only required gspell. I had left in gtkspell by mistake, sorry! Removed.